python/Ganga/Core/InternalServices/Coordinator.py

```diff
@@ -234,10 +180,10 @@ def enableInternalServices():
     Repository_runtime.bootstrap()
 
     # make sure all required credentials are valid
-    missing_cred = getMissingCredentials()
-    if missing_cred:
-        log.error("The following credentials are still required: %s."
-                  "Make sure you renew them before reactivating this session" % ','.join(missing_cred))
+    invalid_afs = [afsToken for afsToken in credential_store.get_all_matching_type(AfsToken) if not afsToken.is_valid()]
+
+    if invalid_afs:
+        log.error('No valid AFS token was found. Please re-authorise before reactivating this session.')
```

Comment:
I think I agree with @rob-c here - maybe this is the time to introduce Ganga bailing out as soon as the AFS token expires? In theory you could keep running and allow the user to get a new AFS token but I suspect that the only real time this happens is with sessions left running overnight in which case they wouldn't be able to recreate the token,